solutions/src/main/java/ru/sberbank/school/task02/FxConversionServiceImpl.java

```diff
@@ -74,24 +79,62 @@ public BigDecimal convert(@NonNull ClientOperation operation, Symbol symbol, Big
      * @param beneficiary В чью пользу осуществляется округление
      * @return Цена для указанного объема
      */
+
     @Override
     public Optional<BigDecimal> convertReversed(@NonNull ClientOperation operation,
-                                         Symbol symbol,
-                                         BigDecimal amount,
-                                         Beneficiary beneficiary) {
+                                                @NonNull Symbol symbol,
+                                                @NonNull BigDecimal amount,
+                                                @NonNull Beneficiary beneficiary) {
         if (amount.compareTo(BigDecimal.ZERO) <= 0) {
             throw new IllegalArgumentException("Wrong amount");
         }
+        List<Quote> rightQuotes = createRightQuotes(exQuotes.getQuotes(symbol), amount, symbol, operation);
+        if (rightQuotes.isEmpty()) {
+            return Optional.empty();
+        }
+        BigDecimal revCur = operation == ClientOperation.BUY ?
+                rightQuotes.iterator().next().getOffer() : rightQuotes.iterator().next().getBid();
+        if (rightQuotes.size() > 1) {
+            revCur = BeneficiaryDepend(beneficiary, rightQuotes, revCur, operation);
+        }
+        return Optional.ofNullable(BigDecimal.ONE.divide(revCur,10,RoundingMode.HALF_UP));
+    }
 
-        ClientOperation revertedOperation = operation == ClientOperation.BUY ? ClientOperation.SELL
-                : ClientOperation.BUY;
-        BigDecimal rightCur = convert(revertedOperation, symbol, amount);
-        BigDecimal rightAmount = amount.divide(rightCur, 10, RoundingMode.HALF_UP);
-        BigDecimal revCur = convert(revertedOperation, symbol, rightAmount);
+    protected BigDecimal BeneficiaryDepend(Beneficiary beneficiary, List<Quote> quotes,
```

Comment:
Method name &#39;BeneficiaryDepend&#39; must match pattern &#39;^[a-z][a-z0-9][a-zA-Z0-9_]*$&#39;.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a423698777>*